src/main/java/opendota/Parse.java

```diff
@@ -496,29 +508,55 @@ public void onTickStart(Context ctx, boolean synthetic) {
         }
     }
 
-    private List<String> getHeroInventory(Context ctx, Entity eHero) {
-        List<String> inventoryList = new ArrayList<>(6);
-        StringTable stEntityNames = ctx.getProcessor(StringTables.class).forName("EntityNames");
-        Entities entities = ctx.getProcessor(Entities.class);
+    private List<Item> getHeroInventory(Context ctx, Entity eHero) {
+        List<Item> inventoryList = new ArrayList<>(6);
 
         for (int i = 0; i < 6; i++) {
-            Integer hItem = eHero.getProperty("m_hItems." + Util.arrayIdxToString(i));
-            if(hItem != 0xFFFFFF) {
-                Entity eItem = entities.getByHandle(hItem);
-                if(eItem == null) {
-                    return null;
-                }
-                String itemName = stEntityNames.getNameByIndex(eItem.getProperty("m_pEntity.m_nameStringableIndex"));
-                if(itemName == null) {
-                    return null;
+            try {
+                Item item = getHeroItem(ctx, eHero, i);
+                if(item != null) {
+                    inventoryList.add(item);
                 }
-                inventoryList.add(itemName);
+            } catch (UnknownItemFoundException e) {
+                return null;
             }
         }
 
         return inventoryList;
     }
 
+    /**
+     * Uses "EntityNames" string table and Entities processor
+     * @param ctx Context
+     * @param eHero Hero entity
+     * @param idx 0-5 - inventory, 6-8 - backpack, 9-16 - stash
+     * @return {@code null} - empty slot. Throws @{@link UnknownItemFoundException} if item information can't be extracted
+     */
+    private Item getHeroItem(Context ctx, Entity eHero, int idx) throws UnknownItemFoundException {
+        StringTable stEntityNames = ctx.getProcessor(StringTables.class).forName("EntityNames");
+        Entities entities = ctx.getProcessor(Entities.class);
+
+        Integer hItem = eHero.getProperty("m_hItems." + Util.arrayIdxToString(idx));
+        if (hItem == 0xFFFFFF) {
+            return null;
+        }
+        Entity eItem = entities.getByHandle(hItem);
+        if(eItem == null) {
+            throw new UnknownItemFoundException(String.format("Can't find item by its handle (%d)", hItem));
+        }
+        String itemName = stEntityNames.getNameByIndex(eItem.getProperty("m_pEntity.m_nameStringableIndex"));
+        if(itemName == null) {
+            throw new UnknownItemFoundException("Can't get item name from EntityName string table");
+        }
+
+        Item item = new Item();
+        item.id = itemName;
+        item.num_charges = eItem.getProperty("m_iCurrentCharges");
```

Comment:
as an optimization maybe don't set these properties if they're 0?

It would make the output slightly less verbose